test/UnitTest.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using Xunit;
+
+namespace SecretHubTest
+{
+    public class TestSuite
+    {
+        [Fact]
+        public void TestReadSuccess()
+        {
+            SecretHub.SecretVersion secret = SecretHub.Client.Read("secrethub-xgo/dotnet/test-secret");
+            Assert.Equal("super_secret_value", secret.Data);

```

Comment:
We can, if we hardcode. The problem is that I have to compare each field seperately, since I cannot create an `expectedSecret`. Some fields are read-only, therefore I cannot set them. If there is a possibility to make all of them have a setter as well, then there are high chances to compare the objects directly.
For now, I leave the test with assertions for all fields.

Reply:
I see the issue. This is good enough for now. We can consider adding setters to all the fields but I do not see any use-case for them besides testing.